constraints_file.py

```diff
@@ -65,25 +65,21 @@ def add_limit_employees_working_days_constraint(shifts: list[Shift], employees:
         constraint_model.Add(sum(shifts_employee_is_working) <= max_working_days)
 
 
-def add_minimum_time_between_closing_shift_and_next_shift_constraint(shifts: list[Shift], employees: list[Employee], constraint_model: cp_model.CpModel, shift_combinations: dict[ShiftCombinationsKey, IntVar], min_time_between_shifts: datetime.timedelta) -> None:
-    closing_shifts = [shift for shift in shifts if shift.shift_type == ShiftTypesEnum.CLOSING]
+def add_minimum_time_between_a_morning_shift_and_the_shift_before_constraint(shifts: list[Shift], employees: list[Employee], constraint_model: cp_model.CpModel, shift_combinations: dict[ShiftCombinationsKey, IntVar], min_time_between_shifts: datetime.timedelta, early_morning_start_time: datetime.time, afternoon_start_time: datetime.time):
+    afternoon_shifts = [shift for shift in shifts if datetime.time(16) <= shift.start_time.time()]
 
     for employee in employees:
+        for afternoon_shift in afternoon_shifts:
+            worked_an_afternoon_shift_yesterday = constraint_model.NewBoolVar(f"afternoon_{afternoon_shift.shift_id}_{employee.employee_id}")
+            afternoon_shift_key = ShiftCombinationsKey(employee.employee_id, afternoon_shift.shift_id)
+            employee_assignment_an_afternoon_shift = worked_an_afternoon_shift_yesterday == shift_combinations[afternoon_shift_key]
 
-        for closing_shift in closing_shifts:
-            worked_closing_shift_yesterday = constraint_model.NewBoolVar(f"closing_{closing_shift.shift_id}_{employee.employee_id}")
-
-            closing_shift_key = ShiftCombinationsKey(employee.employee_id, closing_shift.shift_id)
-            # A variable for better visualization, this represents the assignment to
-            # worked_closing_shift_yesterday BoolVar and not equality. If the IntVar is true (meaning the employee worked),
-            # worked_closing_shift_yesterday will hold true, and vice versa.
-            employee_assignment_closing_shift = worked_closing_shift_yesterday == shift_combinations[closing_shift_key]
-            constraint_model.Add(employee_assignment_closing_shift)
+            constraint_model.Add(employee_assignment_an_afternoon_shift)
 
             forbidden_shifts = [shift_combinations[ShiftCombinationsKey(employee.employee_id, shift.shift_id)] for shift in shifts if
```

Comment:
its hard to understand why you are doing this in list comprehension. build `forbidden_shifts` in a for loop. and give useful names for all the booleans (explaining together why the shifts are forbidden)